fcrepo/fcrepo
# Enforce server managed predicates and types, add interaction model to containment index.

**JIRA Ticket**: https://jira.lyrasis.org/browse/FCREPO-3258, also continues work on https://jira.lyrasis.org/browse/FCREPO-3223

# What does this Pull Request do?
Enforces not being able to change the interaction model of a resource via
* Link header
* RDF body
* PATCH request body

# How should this be tested?

Lots of tests have been added, but now if you add RDF with a `rdf:type` in the namespace of Memento, LDP or Fedora you should get rejected. If you send a Link header different from the current interaction model you should get rejected. If you send a `rdf:type` that is not a URI you should get rejected.

# Interested parties
@fcrepo/committers
